src/query/models/matcher.go

```diff
@@ -108,3 +110,49 @@ func (m Matchers) String() string {
 
 	return buffer.String()
 }
+
+// TODO: make this more robust, handle types other than MatchEqual
+func matcherFromString(s string) (Matcher, error) {
+	ss := strings.Split(s, ":")
+	l := len(ss)
+	if len(ss) > 2 {
+		return Matcher{}, errors.New("invalid arg length for matcher")
+	}
+	fmt.Println(l)
+
+	if l == 0 || len(ss[0]) == 0 {
+		return Matcher{}, errors.New("empty matcher")
+	}
+
+	if l == 1 {
+		return Matcher{
+			Type:  MatchEqual,
```

Comment:
this is a bit weird since you can in theory also have match not equal, match regex, etc.

Reply:
Addressed in the TODO; this is a bit annoying to do since it requires query parsing on our side